services/web/server/src/simcore_service_webserver/projects/projects_handlers.py

```diff
@@ -10,48 +10,57 @@
 
 from aiohttp import web
 from models_library.projects_state import ProjectState
+from pydantic import BaseModel
+from servicelib.aiohttp.requests_validation import (
+    parse_request_path_parameters_as,
+    parse_request_query_parameters_as,
+)
 from servicelib.aiohttp.web_exceptions_extension import HTTPLocked
 from servicelib.json_serialization import json_dumps
+from servicelib.mime import APPLICATION_JSON
 
 from .._meta import api_version_prefix as VTAG
 from ..director_v2_core import DirectorServiceError
-from ..login.decorators import RQT_USERID_KEY, login_required
+from ..login.decorators import login_required
 from ..resource_manager.websocket_manager import PROJECT_ID_KEY, managed_resource
 from ..security_decorators import permission_required
 from . import projects_api
 from .projects_exceptions import ProjectNotFoundError
-from .projects_handlers_crud import routes
+from .projects_handlers_crud import BaseRequestContext, ProjectPathParams
 
 log = logging.getLogger(__name__)
 
+
+routes = web.RouteTableDef()
+
 #
-# Singleton per-user resources https://google.aip.dev/156
+# singleton: Active project
+#  - Singleton per-session resources https://google.aip.dev/156
 #
 
 
-@routes.get(f"/{VTAG}/projects/active")
+class _ProjectActiveParams(BaseModel):
+    client_session_id: str
+
+
+@routes.get(f"/{VTAG}/projects/active", name="get_active_project")
 @login_required
 @permission_required("project.read")
 async def get_active_project(request: web.Request) -> web.Response:
-    user_id: int = request[RQT_USERID_KEY]
-
-    try:
-        client_session_id = request.query["client_session_id"]
+    p = parse_request_query_parameters_as(_ProjectActiveParams, request)
```

Comment:
Q:
1. is user_id is always going to be in there?
2. I absolutely need to have a BaseModel I guess so that you know which query parameter to fetch right?
3. should these model not go in a separate models module?

Reply:
1. only for those handlers with authentication
2. not "absolutely". A possible iteration is to do it like fastapi, i.e. pass all parameters as arguments of the handler function. Then a decorator can parse the signature and produce automatically the model instead of doing it by hand. I did not want to do this right now to avoid many changes at once but I willd efinitively try it
3. as I mention above, I would advocate to move the request body and response payload models to another module but not the parameters (both query and path parameters).